platform/security/idp/security-idp-plugin/security-idp-plugin-api/src/main/java/org/codice/ddf/security/idp/plugin/SamlPresignPlugin.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.idp.plugin;
+
+import ddf.security.samlp.SamlProtocol.Binding;
+import java.util.List;
+import java.util.Set;
+import org.opensaml.saml.saml2.core.AuthnRequest;
+import org.opensaml.saml.saml2.core.Response;
+
+/**
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
```

Comment:
❓  Can we elaborate on the use of this API? Are there threading restrictions? Where in the lifecycle does this get called? Every request?

Reply:
I can do that.